swap/swap.go

```diff
@@ -379,6 +406,41 @@ func (s *Swap) handleEmitChequeMsg(ctx context.Context, p *Peer, msg *EmitCheque
 	return err
 }
 
+func (s *Swap) handleConfirmChequeMsg(ctx context.Context, p *Peer, msg *ConfirmChequeMsg) {
+	p.lock.Lock()
+	defer p.lock.Unlock()
+	cheque := msg.Cheque
+
+	if p.getPendingCheque() == nil {
+		p.logger.Warn("ignoring confirm msg, no pending cheque", "confirm message cheque", cheque)
+		return
+	}
+
+	if !cheque.Equal(p.getPendingCheque()) {
+		p.logger.Warn("ignoring confirm msg, unexpected cheque", "got", cheque, "expected", p.getPendingCheque())
+		return
+	}
+
+	err := p.setLastSentCheque(cheque)
+	if err != nil {
+		p.Drop(fmt.Sprintf("persistence error: %v", err))
+		return
+	}
+
+	err = p.setPendingCheque(nil)
+	if err != nil {
+		p.Drop(fmt.Sprintf("persistence error: %v", err))
+		return
+	}
+
+	// since more swap traffic might have occurred since this cheque was already sent, we redo the payment threshold check
```

Comment:
are you sure we need this?

while it's true that swap traffic might have occurred since the last cheque was sent, isn't this traffic accounted for like any other?

Reply:
It is accounted for like any other but without it it takes one additional call to `Add` after the cheque got confirmed to send a cheque even if it already went over the threshold while waiting. That being said it is quite unlike that the extra `Add` would be enough to reach the disconnect threshold so we could probably remove it.

Reply:
if we don't need it, we have to think of the value this adds to the system in relation to maintaining/testing/reading more code.

since the situation seems to be a rare occurrence, it's probably better to have less code to maintain (and risk less errors) in this particular case. so although the code seems fine, i'd remove it for now.

more opinions are welcome

Reply:
I removed it.